### ISSUE TITLE
API improvement

### ISSUE THREAD

**Issue by fabianfrz** [opentracing/opentracing-ruby#13]
in ruby I usually work with blocks and I would like to do something like this:

lib/opentracing/tracer.rb:

``` ruby
OpenTracing.start_span("parent") do |span|
  # some actions
  OpenTracing.start_span("child", child_of: span) do |child|
    # some more nested actions
  end
  #some actions
end
```
or
``` ruby
OpenTracing.start_span("parent") { |span| do_something_that_needs_to_be_traced }
```

you can check if a block has been used by calling the method `block_given?` and if it returns `true`, `yield` can be called. With this syntax it is easy not to forget the `finish` call.

**Comment by cmur2** [comment]
I created a [helper method like this in my software](https://github.com/cmur2/dyndnsd/blob/147071da9e29cc73e9c111833699ea22da08e561/lib/dyndnsd/helper.rb#L30-L39) since this is lacking from the API.

**Comment by indrekj** [comment]
Seems good idea. Need to also watch out for the return value similarly to #41

**Comment by ethanjcohen-underarmour** [comment]
Just want to point out that `start_active_span` already supports blocks, and can be used in most situations from my experience.

**Comment by mwear** [comment]
Addressed in #45 and released in [opentracing 0.5.0](https://rubygems.org/gems/opentracing/versions/0.5.0).
